### DISCUSSION TITLE
[Angular 進階議題]使用HttpInterceptort為HttpClient Request打點前後大小事 | 全端開發人員天梯

### DISCUSSION THREAD

**Discussion by giscus** [wellwind/wellwind.github.io#74] [Announcements]
# [Angular 進階議題]使用HttpInterceptort為HttpClient Request打點前後大小事 | 全端開發人員天梯

在實際進行專案的時候，Angular內建的HttpClient其實常常是不符合需求的，因為我們可能會需要為每次的HttpRequest都加上一樣的Header，或針對HttpResponse要有一致性的處理，如果同樣的動作在每次Request都進行的話，只會產生一堆同樣的程式碼，既不美觀也容易出錯，好在Angular提供了HttpInterceptor來幫助我們在Reqquest前主動幫我們處理好各種事情！

https://fullstackladder.dev/blog/2017/10/29/angular-advanced-handle-http-request-with-interceptor/

**Comment by thomas-chu-30** [comment]
感謝此篇的分享，我更了解 angular 的 interceptor 了

**Reply by wellwind** [reply]
很高興對你有幫助，有任何問題或建議歡迎隨時提出來一起討論喔 ^^

**Comment by weber87na** [comment]
Authotization 好像拼錯了吧 ..
應該是 Authorization = . =

**Reply by wellwind** [reply]
感謝提醒，已修正囉
